internal/plugin/connectors/tcp/mssql/mock/options.go

```diff
@@ -0,0 +1,103 @@
+package mock
+
+import (
+	"context"
+	"net"
+
+	logmock "github.com/cyberark/secretless-broker/pkg/secretless/log/mock"
+
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql/types"
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+// MSSQLConnectorCtorOptions represents options when creating a MSSQLConnectorCtor. This
+// makes it possible to customize some key values in a Connect double:
+// func Connect(...) (net.Conn, error)
+type MSSQLConnectorCtorOptions struct {
+	// Backend connection returned from the call to Connect
+	BackendConn net.Conn
+	// Error returned from the call to Connect
+	Err error
+	// PreloginResponse from the server passed to interceptor during Connect
+	ServerPreloginResponse map[uint8][]byte
+	// Pointer to unload client login request to, taken from interceptor during Connect
+	ClientLoginRequestPtr **mssql.LoginRequest
+	// LoginResponse from the server passed to interceptor during Connect
+	ServerLoginResponse *mssql.LoginResponse
+}
+
+// MSSQLConnectorCtorOption is the 'functional option' complement to
+// MSSQLConnectorCtorOptions.
+type MSSQLConnectorCtorOption func(*MSSQLConnectorCtorOptions)
+
+// MSSQLConnectorCtor is a 'functional option' of types.ConnectorOption. It generates
+// a customizable MSSQLConnectorCtor using MSSQLConnectorCtorOptions, then sets it on the
+// ConnectorOption passed to it.
+//
+// Empty values of MSSQLConnectorCtorOption result default:
+// ServerPreloginResponse => map[uint8][]byte{}
+// ServerLoginResponse => &mssql.LoginResponse{}
+// BackendConn => nil
+// BackendConn => nil
```

Comment:
Quote @diverdane 
> Do we want to use 'options' here instead of 'setters', per Johah's previous comment, or is this used intentionally to distinguish this from ConnectorOptions?